sklearn/utils/seq_dataset.pyx.tp

```diff
@@ -161,17 +195,18 @@ cdef class SequentialDataset:
 
     def _sample_py(self, int current_index):
         """python function used for easy testing"""
-        cdef double* x_data_ptr
```

Comment:
I am unsure about this function for testing.

`double` and `float` in C will be converted to `float` in python. Therefore, we cannot test that we are having the good type so the function `*_py` are not useful for type checking I think.